dsgrid/registry/project_registry_manager.py

```diff
@@ -355,6 +396,145 @@ def _register_dimensions_from_models(
+                    records["id"].append(selector.name)
+                    records["name"].append(selector.description)
+                    if selector.column_values:
+                        for column, value in selector.column_values.items():
+                            records[column].append(value)
+                filename = tmp_path / f"{subset_dimension.dimension_query_name}.csv"
+                pd.DataFrame(records).to_csv(filename, index=False)
+
+                dim = DimensionModel(
+                    filename=str(filename),
+                    name=subset_dimension.name,
+                    display_name=subset_dimension.display_name,
+                    dimension_type=dimension_type,
+                    module=base_dim.model.module,
+                    class_name=base_dim.model.class_name,
+                    description=subset_dimension.description,
+                )
+                dimensions.append(dim)
+
+        dim_model = DimensionsConfigModel(dimensions=dimensions)
+        dims_config = DimensionsConfig.load_from_model(dim_model)
+        dimension_ids = self._dimension_mgr.register_from_config(
+            dims_config, submitter, log_message, context=context
+        )
+        model.dimensions.supplemental_dimension_references += (
+            self._dimension_mgr.make_dimension_references(dimension_ids)
+        )
+
+    def _add_supplemental_dimension_mappings_from_subset_dimensions(
+        self, src_dir, supp_dir, model
+    ):
+        new_mappings = []
+        for subset_dimension in model.dimensions.subset_dimensions:
+            dimension_type = subset_dimension.dimension_type
+            base_dim = None
+            for ref in model.dimensions.base_dimension_references:
+                if ref.dimension_type == dimension_type:
+                    base_dim = self._dimension_mgr.get_by_id(ref.dimension_id)
+                    break
+            assert base_dim is not None, subset_dimension
+            base_dim_record_ids = base_dim.get_unique_ids()
+            mapping_records = []
+            new_ids = set()
+            for selector in subset_dimension.selectors:
+                for record_id in selector.records:
+                    mapping_records.append({"from_id": record_id, "to_id": selector.name})
+                    new_ids.add(record_id)
+            for record_id in base_dim_record_ids.difference(new_ids):
+                mapping_records.append({"from_id": record_id, "to_id": ""})
+            map_record_file = supp_dir / f"{subset_dimension.dimension_query_name}_mapping.csv"
+            pd.DataFrame.from_records(mapping_records).to_csv(map_record_file, index=False)
+
+            with in_other_dir(src_dir):
+                mapping = MappingTableByNameModel(
+                    filename=str(Path(_SUPPLEMENTAL_TMP_DIR) / map_record_file.name),
+                    mapping_type=DimensionMappingType.MANY_TO_MANY_EXPLICIT_MULTIPLIERS,
```

Comment:
I’ll defer to @lixiliu

Reply:
@lixiliu, can you touch base with @daniel-thom on this this week?

Reply:
MANY_TO_MANY_EXPLICIT_MULTIPLIERS is correct because we are not guaranteeing that the fractions (which are all ones) add up to 1.0 in any direction (either over from_id or to_id).